### PR TITLE
AI Extension: handle Jetpack Form button instances when composing via AI assistant

### LINKED ISSUE #32562: AI Extension: form generated on FSE may not contain the submit button
While generating forms on FSE, the resulting form may not contain the submit button:

Image

The markup generated is correct, so it may be an issue with the specific FSE editor context:

Image

May be related to #32561

### PR DESCRIPTION
**Pull request by retrofox**
This PR checks the number of Button instances when the Jetpack Form is created by using the AI assistant:

* - if it exists twice or more, remove all less the last one
* - if it does not exist, create one.

Fixes https://github.com/Automattic/jetpack/issues/32562

## Proposed changes:

* AI Extension: handle Jetpack Form button instances when composing via AI assistant

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

It is challenging to test because usually, the extension handles the Button block instances correctly. What I did was [commenting this line](https://github.com/Automattic/jetpack/blob/901c2dd2c65893545271d5c713c5fae4db969d31/projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/prompt/index.ts#L261), trying to do not to get the Button in the response.

* Go to the block editor
* Create a Jetpack Form block instance
* Ask for a Form without a Button form
* **After**: The button is added when the response doesn't contain the button. You can check that in the dev console.

https://github.com/Automattic/jetpack/assets/77539/8a6b2e0e-d32f-4b61-9807-f9becd07e7cc

* Ask for a Form with many jetpack button blocks
* Ensure the app corrects the instances:

https://github.com/Automattic/jetpack/assets/77539/ef78381f-a6af-4e95-826f-480f2903a1fd

* Confirm that always, in the end, the Form contains only one Button block instance.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx [hunk 1]
```diff
@@ -173,14 +175,50 @@ const withUiHandlerDataProvider = createHigherOrderComponent( BlockListBlock =>
...
+					 * - if it exists twice or more, remove the first one.
+					 * - if it does not exist, create one.
+					 */
+					const allButtonBlocks = validBlocks.filter( block => block.name === 'jetpack/button' );
+					currentListOfValidBlocks.current = currentListOfValidBlocks.current || [];
+					if ( allButtonBlocks.length > 1 ) {
+						// Remove all button blocks, less the last one.
+						currentListOfValidBlocks.current = currentListOfValidBlocks.current.filter(
+							( block, i ) => {
+								return block.name !== 'jetpack/button' || i === allButtonBlocks.length - 1;
```
Comment (lhkowalski) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx]: This line seems a bit strange to me. 

Suppose we have 5 fields in the form and then 2 buttons at the end. The `i` value for them would be 5 and 6, right? Also, `allButtonBlocks` will have 2 elements, so `allButtonBlocks.length` will be 2. In that case, `i === allButtonBlocks.length - 1` will not be true for a button, but for a random field on the list.

Idea: what about filtering out all the buttons, and then appeding the last item of `allButtonBlocks` to the list?

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx]: You're right. Let me update it

Comment (lhkowalski) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx]: Just to confirm, I tested it asking for 2 fields and 2 buttons, and I end up with no buttons, despite the buttons being on the completion response.

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx]: https://github.com/Automattic/jetpack/pull/32581/commits/3609a0d3c21fb95fbc119d3b8ce6fedf8698e914


https://github.com/Automattic/jetpack/assets/77539/1cc6b6be-7a46-4411-88ad-b34a51daeb15

**Review by lhkowalski — APPROVED** [review]
Code looks good now, and it's working the way it should.

If there is more than 1 button, only one is included (in this case I asked for more buttons for testing purposes).

The completion content:

Screenshot 2023-08-18 at 14 39 45

A readable version:

Screenshot 2023-08-18 at 14 40 12

And the resulting form:

Screenshot 2023-08-18 at 14 39 28

---

If there is no button, one is included automatically (in this case I did the prompt tweak to not ask for the button).

The completion content:

Screenshot 2023-08-18 at 14 45 16

A readable version:

Screenshot 2023-08-18 at 14 45 34

And the resulting form:

Screenshot 2023-08-18 at 14 44 56

Nice job!

**Pull request merged**